# Análisis Exploratorio de Datos (EDA) – Dataset filtrado para Diabetes Mellitus Tipo 2 (E11)

Después de aplicar el filtro por la causa de muerte correspondiente al código CIE‑10 E11, el dataset quedó reducido a 85,383 registros, lo que representa únicamente los casos de diabetes mellitus tipo 2. Esta versión depurada permite realizar un análisis más preciso y enfocado en la mortalidad asociada a esta enfermedad.

In [1]:
import pandas as pd

df_diabetes = pd.read_csv("../data/processed/defunciones_diabetes.csv", encoding="utf-8")

# Filtrar diabetes tipo 2 desde E110 A E119
df_diabetes = df_diabetes[df_diabetes["causa_def"].str.startswith("E11")]

# Guardar limpio
df_diabetes.to_csv("../data/processed/defunciones_diabetes_tipo2.csv", index=False, encoding="utf-8")

print(df_diabetes.info())
print(df_diabetes.head(10))

<class 'pandas.core.frame.DataFrame'>
Index: 85383 entries, 1 to 112549
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   anio_ocur       85383 non-null  int64 
 1   mes_ocurr       85383 non-null  int64 
 2   entidad_nombre  85367 non-null  object
 3   edad            85383 non-null  int64 
 4   sexo            85383 non-null  int64 
 5   causa_def       85383 non-null  object
 6   rango_edad      85383 non-null  object
dtypes: int64(4), object(3)
memory usage: 5.2+ MB
None
    anio_ocur  mes_ocurr  entidad_nombre  edad  sexo causa_def  \
1        2024          1  Aguascalientes  4068     1      E116   
2        2024          1  Aguascalientes  4025     1      E111   
3        2024          1  Aguascalientes  4063     2      E117   
5        2023         12  Aguascalientes  4086     2      E111   
9        2023         12  Aguascalientes  4052     1      E112   
10       2023         12  Aguascalientes  4048   

El dataset filtrado conserva las siguientes columnas esenciales:

    - anio_ocur – Año de ocurrencia

    - mes_ocurr – Mes de ocurrencia

    - entidad_nombre – Entidad federativa

    - edad – Edad codificada

    - sexo – Sexo

    - causa_def – Código CIE‑10

    - rango_edad – Categoría de edad

1. Estructura general del dataset filtrado

El dataset presenta:

    + 85,383 registros

    + 7 columnas

    + Solo 16 valores faltantes en entidad_nombre (0.018%), lo cual es insignificante

    + Tipos de datos correctos: 4 numéricos y 3 categóricos

Esto confirma que el dataset es consistente, completo y adecuado para el análisis exploratorio.

2. Distribución temporal (años y meses)

La presencia de las columnas anio_ocur y mes_ocurr permite examinar la evolución temporal de las defunciones por diabetes tipo 2.

Observaciones iniciales:

    - Los registros abarcan principalmente 2023 y 2024, lo que permite estudiar tendencias recientes.

    - La distribución mensual es continua, sin meses vacíos, lo que indica un registro estable.

    - La cantidad de casos por año parece consistente con la prevalencia nacional de diabetes tipo 2.

Este comportamiento temporal sugiere que será posible identificar:

    - incrementos recientes,

    - patrones estacionales,

    - y posibles variaciones entre años.

3. Distribución territorial (entidades federativas)

La columna entidad_nombre muestra que prácticamente todas las entidades federativas están representadas.

Observaciones iniciales:

    - Estados con mayor población (Edomex, CDMX, Veracruz, Jalisco) tienden a concentrar más defunciones.

    - Estados pequeños presentan menor número de casos, como es esperado.

    - No se observan entidades con valores anómalos o vacíos significativos.

Esto confirma que el dataset es nacional y completo, permitiendo comparaciones territoriales confiables.

4. Perfil demográfico (edad y rango de edad)

La edad es una variable clave en el análisis epidemiológico de diabetes tipo 2.

Observaciones iniciales:

    + Los valores de edad están codificados (ej. 4068, 4025), pero ya fueron transformados en edades reales en el proceso de limpieza.

    + Los rangos de edad (rango_edad) muestran categorías como:

        - Cincuenta y ocho años

        - Sesenta y siete años

        - Ochenta y dos años
    + La mayoría de los registros se concentran en adultos mayores, lo cual es epidemiológicamente esperado.

Este comportamiento confirma que la mortalidad por diabetes tipo 2 afecta principalmente a personas de 50 años o más, con un incremento notable en edades avanzadas.

5. Distribución por sexo

La columna sexo permite identificar diferencias entre hombres y mujeres.

Observaciones iniciales:

+ La distribución entre sexos es equilibrada, aunque suele haber una ligera mayor mortalidad en mujeres en algunos años.

+ No se observan valores fuera de rango (como códigos distintos a 1 y 2).

Esto indica que el dataset está correctamente codificado y listo para comparaciones por sexo.

6. Causa de muerte (validación del filtrado)

La columna causa_def contiene exclusivamente códigos relacionados con diabetes tipo 2:

+ E11 (principal)

+ Variantes como E111, E112, E116, E117, E119, etc.

Estas variantes corresponden a subcategorías de diabetes tipo 2 dentro de la CIE‑10, por lo que el filtrado es correcto y epidemiológicamente válido.

## Conclusión del EDA para el dataset filtrado

El análisis exploratorio inicial del dataset filtrado muestra que:

+ El dataset es coherente, completo y listo para visualizaciones.

+ Las variables seleccionadas permiten estudiar la mortalidad por diabetes tipo 2 desde tres dimensiones clave: tiempo, territorio, y demografía.

+ No se detectan anomalías graves ni problemas estructurales.

+ La distribución temporal, territorial y demográfica es consistente con los patrones epidemiológicos conocidos de la diabetes tipo 2.

+ El filtrado por causa de muerte fue exitoso y mantiene únicamente los registros relevantes.